src/PostgREST/Request/ApiRequest.hs

```diff
@@ -113,9 +113,10 @@ toRpcParamValue proc (k, v) | argIsVariadic k = (k, Variadic [v])
   where
     argIsVariadic arg = isJust $ find (\PgArg{pgaName, pgaVar} -> pgaName == arg && pgaVar) $ pdArgs proc
 
--- | Convert rpc params `/rpc/func?a=val1&b=val2` to json `{"a": "val1", "b": "val2"}
+-- | Convert rpc params `/rpc/func?a::type=val1&b=val2` to json `{"a": "val1", "b": "val2"}
 jsonRpcParams :: ProcDescription -> [(Text, Text)] -> PayloadJSON
-jsonRpcParams proc prms =
+jsonRpcParams proc castedParams =
+  let prms = first (fst . T.breakOn "::") <$> castedParams in
```

Comment:
The casted params are converted to a json "payload"  without the explicit casting from the client request.